tests/pyrog/test_api.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from faker import Faker
+
+from django.urls import reverse
+
+faker = Faker()
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("name, version, status_code", [(faker.word(), faker.word(), 201), ("", faker.word(), 400)])
+def test_create_source(api_client, template, name, version, status_code):
+    url = reverse("sources-list")
```

Comment:
where do `sources-list` and `soureces-detail` come from ?

Reply:
This is well-defined in DRF (cf https://www.django-rest-framework.org/api-guide/routers/#simplerouter)
A Router takes a ViewSet and creates a set of URL routes mapping to the CRUD operations.
In django, URL routes have names. Using `reverse` allows to find the associated `path` (cf https://docs.djangoproject.com/en/3.1/ref/urlresolvers/#reverse).